contents/tutorials/remix-analytics.md

```diff
@@ -0,0 +1,520 @@
+
+  const fileContents = await fs.readFile(filePath, 'utf8');
+  const posts: Post[] = JSON.parse(fileContents).posts;
+
+  const post = posts.find(p => p.slug === params.slug);
+
+  if (!post) {
+    throw new Response("Not Found", { status: 404 });
+  }
+
+  return json(post);
+};
+
+export default function PostSlug() {
+  const post = useLoaderData<Post>();
+
+  const handleClick = () => {
+    console.log('liked')
+  };
+
+  return (
+    <div>
+      <h1>{post.title}</h1>
+      <p>{post.content}</p>
+      <button onClick={handleClick}>Like this post</button>
+    </div>
+  );
+}
+```
+
+Finally, go into the project folder and run the dev server with the following command:
+
+```bash
+cd remix-tutorial
+npm run dev
+```
+
+ Once done, your basic Remix blog is running locally.
+
+![App](../images/tutorials/remix-analytics/app.png)
+
+## Adding PostHog on the client side
+
+Up next is adding PostHog, which you can start by installing:
+
+```bash
+npm i posthog-js
+```
+
+Once done, you need your project API key and instance address from your PostHog [project settings](https://app.posthog.com/settings/project). You can [sign up for free](https://app.posthog.com/signup) if you haven’t already.
+
+With these, go to `routes/root.tsx` and set up a `useEffect` to initialize PostHog (import both). 
+
+```ts
+// routes/root.tsx
+import { cssBundleHref } from "@remix-run/css-bundle";
+import type { LinksFunction } from "@remix-run/node";
+import {
+  Links,
+  LiveReload,
+  Meta,
+  Outlet,
+  Scripts,
+  ScrollRestoration
+} from "@remix-run/react";
+import posthog from "posthog-js"; // new
+import { useEffect } from "react"; // new
+
+export const links: LinksFunction = () => [
+  ...(cssBundleHref ? [{ rel: "stylesheet", href: cssBundleHref }] : []),
+];
+
+export default function App() {
+  useEffect(() => {
+    posthog.init(
+      '<ph_project_api_key>',
+      {
+        api_host:'<ph_instance_address>'
+      }
+    )
+  }
+  , [])
+
+  return (
+    <html lang="en">
+      <head>
+        <meta charSet="utf-8" />
+        <meta name="viewport" content="width=device-width, initial-scale=1" />
+        <Meta />
+        <Links />
+      </head>
+      <body>
+        <Outlet />
+        <ScrollRestoration />
+        <Scripts />
+        <LiveReload />
+      </body>
+    </html>
+  );
+}
+```
+
+After relaunching your app, PostHog begins autocapturing initial pageviews, clicks, [session replays](/docs/session-replay) (if [you enable them](https://app.posthog.com/settings/project#replay)), and more.
+
+![Autocapture](../images/tutorials/remix-analytics/autocapture.png)
+
+## Capturing pageviews
+
+You might notice we captured only one pageview even though we navigated between multiple pages. This is because Remix acts as a [single-page app](/tutorials/single-page-app-pageviews) and only triggers an initial page load event. PostHog uses the page load event for pageviews, so to fix this, we must implement pageview capture ourselves.
```

Comment:
```suggestion
You might notice we captured only one pageview even though we navigated between multiple pages. This is because Remix is a [single-page app](/tutorials/single-page-app-pageviews) and only triggers an initial page load event. PostHog uses the page load event for pageviews, so to fix this, we must implement pageview capture ourselves.
```